crates/valence_protocol/src/ident.rs

```diff
@@ -40,6 +40,17 @@ pub struct Ident<S> {
     path_start: usize,
 }
 
+impl<S> Ident<S> {
+    /// Returns an Ident with the given fields
+    ///
+    /// # Safety
+    /// This function does not check for the validity of the Ident.
+    /// For a safe version use [`Ident::new`]
+    pub const unsafe fn new_unchecked(string: S, path_start: usize) -> Self {
```

Comment:
I don't think this function should be marked as unsafe because unsafe code cannot rely on the invariants of `Ident` to be upheld to begin with. By using an evil implementation of `AsRef` or `Borrow` combined with `as_str_ident`, you can create invalid `Ident`s within safe code.

That said, this function really should not be used under normal circumstances. It would be good to add `#[doc(hidden)]`.